docs/spec_draft.md

```diff
@@ -1443,6 +1444,138 @@ specification. Numeric precision is implementation-defined.
+| `operand`           | tensor of any supported type              | (C1-C10)         |
+| `source`            | tensor of any supported type              | (C1-C3)          |
+| `init_value`        | scalar of any supported type              | (C4), (C9)       |
+| `window_dimensions` | 1-dimensional tensor constant type `si64` | (C1), (C3), (C5) |
+| `window_strides`    | 1-dimensional tensor constant type `si64` | (C6)             |
+| `padding`           | 2-dimensional tensor constant type `si64` | (C3), (C7)       |
+| `select`            | `function`                                | (C8)             |
+| `scatter`           | `function`                                | (C4), (C9)       |
+
+### Outputs
+
+| Name     | Type                         | Constraints |
+|----------|------------------------------|-------------|
+| `result` | tensor of any supported type | (C9-C10)    |
+
+### Constraints
+
+  * (C1) rank(`source`) $=$ rank(`operand`) $=$ size(`window_dimensions`).
+
+  * (C2) `operand` and `source` have the same element type ignoring floating-point
+    precision.
+
+  * (C3) `dim(source, i) = strided_bound[i]` for any i $\in$ axis(`source`)
+    * `strided_bound[i] = (padded_operand[i] == 0 || window_dimensions[i] > padded_operand[i]) ? 0 : floor((padded_operand[i] - window_dimensions[i]) / window_strides[i]) + 1`
+    * `padded_operand[i] = padding[i][0] + dim(operand, i) + padding[i][1]`
+
+  * (C4) `init_value` value has to form an identity under `scatter`.
+         The behavior of the operation is undefined otherwise.
+
+  * (C5) On `window_dimensions`
+    * size(`window_dimensions`) $=$ rank(`operand`).
+    * `window_dimensions[i]` $\gt 0$ for all i $\in$ [0, size(window_dimensions)).
+
+  * (C6) On `window_strides` (optional)
+    * Default value of 1 for each dimension of `operand`.
+    * size(`window_strides`) $=$ rank(`operand`).
+    * `window_strides[i]` $\gt 0$ for all i $\in$ [0, size(window_strides)).
+
+  * (C7) On `padding`(optional)
+    * Default value of (0,0) for each dimension of `operand`.
+    * size(`padding`) $=$ rank(`operand`).
+    * dim(`padding`, 0) $=$ N and dim(`padding`, 1) = 2.
+
+  * (C8) `select` has type `(T, T) -> i1` such that T is a 0-D tensor with
+        element type same as that of 'operand'. It is called with the property
+        that the first parameter's index vector is lexicographically less than
+        the second parameter's index vector. The select function returns true if
+        the first parameter is selected and returns false if the second
+        parameter is selected and the function must hold transitivity
+        (i.e., if select(a, b) and select(b, c) are true, then select(a, c) is
+        also true) so that the selected element does not depend on the order of
+        the elements traversed for a given window.
+
+  * (C9) The `scatter` has type `(T, T) -> T, where the first and second input
```

Comment:
Is the order of arguments for `scatter` enforced (aka `operand` first and `init_value` second)?

Reply:
Nice point. Fixed it as pointed out in  [ReduceOp::comment](https://github.com/openxla/stablehlo/pull/353#discussion_r1002758771).